### PR TITLE
[refs #0357] Upgrade CircleCi 2.0

### PR DESCRIPTION
**Pull request by yourownmood**
This upgrades the current CircleCi 1.0 configuration to CircleCi 2.0.

### CODE REVIEW

**Comment by florianbouvot** [comment]
@yourownmood thank you, I've never work with CircleCI.
@inuitcss/core someone to look at this ?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      LANG: en_US.UTF-8
+    docker:
+      - image: circleci/build-image:ubuntu-14.04-XXL-upstart-1189-5614f37
+        command: /sbin/init
+    steps:
+      - checkout
+      - run: mkdir -p $CIRCLE_ARTIFACTS $CIRCLE_TEST_REPORTS
+      - run:
+          working_directory: ~/inuitcss/inuitcss
+          command: nvm install 8.9.3 && nvm alias default 8.9.3
```
Comment (anselmh) [inline .circleci/config.yml]: I think this should use `nvm install 8 && nvm alias default 8` instead in order to get bugfixes and minor versions automatically.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    working_directory: ~/inuitcss/inuitcss
+    parallelism: 1
+    shell: /bin/bash --login
+    environment:
+      CIRCLE_ARTIFACTS: /tmp/circleci-artifacts
+      CIRCLE_TEST_REPORTS: /tmp/circleci-test-results
+      LC_ALL: en_US.UTF-8
+      LANG: en_US.UTF-8
+    docker:
+      - image: circleci/build-image:ubuntu-14.04-XXL-upstart-1189-5614f37
```
Comment (anselmh) [inline .circleci/config.yml:13]: Hm, do we really need such a big image? I’d be in favor of using a smaller distribution like arch or alpine…

**Comment by florianbouvot** [comment]
@yourownmood are you available to update your PR?

**Comment by yourownmood** [comment]
I will try to have a look today 🤞

**Comment by yourownmood** [comment]
I have updated the pr with the comment about the use of the nvm version number.

**About the docker-image (and the size):**
I've tried to get this up and running but i ran into to, too much complexity and extra dependencies we have to manage.
- CircleCi has a list of dependencies that it needs to be able to run.
- The inuitcss project requires node and nvm.
- The inuitcss requires ruby and certain gems.
- To install all those specific dependencies other dependencies are required curl, apk etc.

This would make things more complicated (or we need to maintain a own docker image).

About the size of the image:
Since this is a CircleCi maintained image it also cached on CircleCI which means it "never" needs to download this image.

Let me know your thoughts.

**Review by anselmh — APPROVED** [review]

**Comment by florianbouvot** [comment]
@yourownmood thank you 👍 I think that we could merge this.

**Comment by yourownmood** [comment]
After this pr is merged, we could rebase #370 on master.
This will kick-off the "new" CircleCi setup

**Pull request merged**